pkg/wsman/cim/kvm/decoder.go

```diff
@@ -0,0 +1,155 @@
+/*********************************************************************
+ * Copyright (c) Intel Corporation 2023
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+
+package kvm
+
+const CIM_KVMRedirectionSAP string = "CIM_KVMRedirectionSAP"
+
+const (
+	RedirectionSAP_Enable  KVMRedirectionSAPRequestStateChangeInput = 2
+	RedirectionSAP_Disable KVMRedirectionSAPRequestStateChangeInput = 3
+)
+
+// KVMProtocol constants
+const (
+	KVMProtocolUnknown KVMProtocol = iota
+	KVMProtocolOther
+	KVMProtocolRaw
+	KVMProtocolRDP
+	KVMProtocolVNCRFB
+)
+
+// kvmProtocolToString is a map of KVMProtocol value to string
+var kvmProtocolToString = map[KVMProtocol]string{
+	KVMProtocolUnknown: "Unknown",
+	KVMProtocolOther:   "Other",
+	KVMProtocolRaw:     "Raw",
+	KVMProtocolRDP:     "RDP",
+	KVMProtocolVNCRFB:  "VNC-RFB",
+}
+
+// String returns a human-readable string representation of the KVMProtocol enumeration
+func (e KVMProtocol) String() string {
+	if s, ok := kvmProtocolToString[e]; ok {
+		return s
+	}
+	return "Value not found in map"
+}
+
+// ReturnValue constants
+const (
+	ReturnValueCompletedNoError ReturnValue = iota
+	ReturnValueNotSupported
+	ReturnValueUnknownError
+	ReturnValueTimeout
+	ReturnValueFailed
+	ReturnValueInvalidParameter
+	ReturnValueInUse
+	ReturnValueMethodParametersChecked
```

Comment:
if these values are for CIM_KVMRedirectionSAP RequestStateChange than there are reserved values between "In Use" and "Method Parameters Checked"

Reply:
fixed

Reply:
fixed